File: 009-palindrome number/Solution009.cpp
#include <stdio.h>

//Attempt1: 126ms
class Solution009 {
public:
  bool isPalindrome(int x) {
    if (x < 0)
      return false;

    int p = x;
    long y = 0;
    while (p > 0) {
      y = y * 10 + p % 10;
      p /= 10;
    }

    return x == y;
  }
};

File: 009-palindrome number/Solution009.java
public class Solution009 {
    //Attempt1: 444ms
    public boolean isPalindrome(int x) {
        if (x < 0)
            return false;
        if (x < 10)
            return true;

        int len = (int) Math.log10(x);
        while (len > 0) {
            int pow = (int)Math.pow(10, len);
            if (x / pow != x % 10)
                return false;
            x = (x % pow) / 10;
            len -= 2;
        }

        return true;
    }

    public static void main(String[] args) {
        System.out.println(new Solution009().isPalindrome(10));
    }
}
